cmake/psi4OptionsTools.cmake

```diff
@@ -61,6 +61,7 @@ macro(psi4_add_module binlib libname sources)
   target_link_libraries(${libname}
     PRIVATE
       tgt::lapack
+      Libint2::cxx
```

Comment:
for those reviewing, this is linking L2 into _every_ p4 module. this contrasts with L1 which only linked into mints. Reason is b/c using L2 shell directly in basisset.h, and that's included everywhere. worth a revisit to check absolutely necessary.

Reply:
Excellent news!  I changed the design a lot, and now only eri.cc and eribase.cc include the libint2 headers; it's forward declared elsewhere.  Let's change the behavior back to the old libint1 way and save a lot of compile time!

Reply:
great, kill this line, then back to mints only! let me know if you're pausing this evening or weekend, then I'll bully azure for a bit.

Reply:
have at it!  the lastest pushed version is rebased to master and passes full tests and pytests.  i'm just writing some docs right now, so won't conflict with any of your stuff